homeassistant/components/elmax/strings.json

```diff
@@ -1,13 +1,37 @@
 {
   "config": {
     "step": {
-      "user": {
+      "choose_mode": {
+        "description": "Please choose the connection mode to Elmax panels.",
+        "menu_options": {
+          "cloud": "Connect to Elmax Panel via Elmax Cloud APIs",
+          "direct": "Connect to Elmax Panel via local/direct IP"
+        }
+      },
+      "cloud": {
         "description": "Please login to the Elmax cloud using your credentials",
         "data": {
           "password": "[%key:common::config_flow::data::password%]",
           "username": "[%key:common::config_flow::data::username%]"
         }
       },
+      "zeroconf_setup": {
+        "description": "Configure discovered local Elmax panel",
+        "data": {
+          "panel_pin": "Panel PIN code",
+          "use_ssl": "Use SSL"
+        }
+      },
+      "direct": {
+        "description": "Specify the Elmax panel connection parameters below.",
+        "data": {
+          "panel_api_host": "Panel API Hostname or IP",
+          "panel_api_port": "Panel API port",
+          "use_ssl": "Use SSL",
+          "follow_mdns": "Automatically update HOST/PORT via mdns",
```

Comment:
Please add a `data_description` dictionary with explanations for the use_ssl and follow_mdns options.

Reply:
If the panel gets unlinked in the scenario you pictured, the API should return a 403.[ That is the low-level API will raise an `ElmaxBadPinError` which will trigger the re-auth flow again](https://github.com/albertogeniola/elmax-api/blob/1fba54b87e4defcc4bc0c43422a77414b89220b8/elmax_api/http.py#L592C1-L593C1). I guess that's a reasonable manner to handle this edge-case.

I've also added the `data_description` entries as suggested in 9fef91b5b075d2f85287fbc794c853de8b1d5747. Thanks!

Reply:
I'm not sure you'd like me to comment the following:

> I meant both the SSL and mDNS options.
> If we require user to enter host and port in direct mode, why is mDNS involved?

In any case, the point is that the user might configure the panel in "DIRECT MODE" when the panel is assigned a dynamic IP. If this is the case, the user might want to enable mDNS updating mode. That's why we ask for user's input at this stage about mDNS.

I hope this clarifies!

Reply:
> The point is that the user might configure the panel in "DIRECT MODE" when the panel is assigned a dynamic IP. If this is the case, the user might want to enable mDNS updating mode. That's why we ask for user's input at this stage about mDNS.

OK, I see. But updating the config entry based on mDNS data is common, we do that for many integrations automatically without user intervention. Why do we make it a user configurable option for this integration?

Reply:
I did not realize that was so common... Do you suggest to remove this option right away ?

Reply:
Done in 253a77d858f3d6f2ea3889ad3fea0f1072b00f32